src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -696,6 +701,7 @@ void VideoDecoder::maybeSeekToBeforeDesiredPts() {
   if (!firstStreamInfo.keyFrames.empty()) {
     int desiredKeyFrameIndex =
         getKeyFrameIndexForPts(firstStreamInfo, desiredPts);
+    desiredKeyFrameIndex = std::max(desiredKeyFrameIndex, 0);
```

Comment:
When tracing through this logic, I realized that we're duplicating the check that's inside of `getKeyFrameIndexForPts`. In that function, we'll always call `getKeyFrameIndexForPtsUsingScannedIndex`. It might be better to just call that function directly, as it also matches with the comment above.